src/tracker/camera_control.c

```diff
@@ -130,52 +139,76 @@ camera_control_new(int cameraID)
 	cc->camera = CLEyeCreateCamera(cguid,
-                CLEYE_COLOR_PROCESSED, CLEYE_VGA, 60);
+        CLEYE_COLOR_PROCESSED, CLEYE_VGA, framerate);
 
-	CLEyeCameraGetFrameDimensions(cc->camera, &w, &h);
+    CLEyeCameraGetFrameDimensions(cc->camera, &width, &height);
 
 	// Depending on color mode chosen, create the appropriate OpenCV image
-	cc->frame4ch = cvCreateImage(cvSize(w, h), IPL_DEPTH_8U, 4);
-	cc->frame3ch = cvCreateImage(cvSize(w, h), IPL_DEPTH_8U, 3);
+    cc->frame4ch = cvCreateImage(cvSize(width, height), IPL_DEPTH_8U, 4);
+    cc->frame3ch = cvCreateImage(cvSize(width, height), IPL_DEPTH_8U, 3);
 
 	CLEyeCameraStart(cc->camera);
+
 #elif defined(CAMERA_CONTROL_USE_PS3EYE_DRIVER)
-        ps3eye_init();
-        int cams = ps3eye_count_connected();
+    ps3eye_init();
+    int cams = ps3eye_count_connected();
 
-        if (cams <= cameraID) {
-            free(cc);
-            return NULL;
-        }
+    if (cams <= cameraID) {
+        free(cc);
+        return NULL;
+    }
 
-        get_metrics(&(cc->width), &(cc->height));
+    if (width <= 0 || height <= 0) {
+        get_metrics(&width, &height);
+    }
 
-        cc->eye = ps3eye_open(cameraID, cc->width, cc->height, 60);
+    cc->eye = ps3eye_open(cameraID, width, height, framerate);
 
-        cc->framebgr = cvCreateImage(cvSize(cc->width, cc->height), IPL_DEPTH_8U, 3);
+    if (cc->eye == NULL) {
+        free(cc);
+        return NULL;
+    }
+
+    cc->framebgr = cvCreateImage(cvSize(width, height), IPL_DEPTH_8U, 3);
 #else
-        char *video = psmove_util_get_env_string(PSMOVE_TRACKER_FILENAME_ENV);
+    char *video = psmove_util_get_env_string(PSMOVE_TRACKER_FILENAME_ENV);
 
-        if (video) {
-            psmove_DEBUG("Using '%s' as video input.\n", video);
-            cc->capture = cvCaptureFromFile(video);
-            free(video);
-        } else {
-            cc->capture = cvCaptureFromCAM(cc->cameraID);
+    if (video) {
+        psmove_DEBUG("Using '%s' as video input.\n", video);
+        cc->capture = cvCaptureFromFile(video);
+        free(video);
+    } else {
+        cc->capture = cvCaptureFromCAM(cc->cameraID);
 
-            int width, height;
+        if (width <= 0 || height <= 0) {
             get_metrics(&width, &height);
-
-            cvSetCaptureProperty(cc->capture, CV_CAP_PROP_FRAME_WIDTH, width);
-            cvSetCaptureProperty(cc->capture, CV_CAP_PROP_FRAME_HEIGHT, height);
         }
-#endif
 
-        cc->deinterlace = PSMove_False;
+        cvSetCaptureProperty(cc->capture, CV_CAP_PROP_FRAME_WIDTH, width);
+        cvSetCaptureProperty(cc->capture, CV_CAP_PROP_FRAME_HEIGHT, height);
+    }
+#endif
+    cc->width = width;
+    cc->height = height;
+    cc->deinterlace = PSMove_False;
 
 	return cc;
 }
 
+int
+camera_control_count_connected()
+{
+#if defined(CAMERA_CONTROL_USE_CL_DRIVER)
+	return CLEyeGetCameraCount();
+#elif defined(CAMERA_CONTROL_USE_PS3EYE_DRIVER)
+	ps3eye_init();
```

Comment:
Actually, thinking about it some more, it might be a good idea to move `ps3eye_init` out of these functions altogether and perhaps move them to a (new) `psmovetracker_init` function? Makes it more explicit and less lazy. 

On the other hand, it would be a pretty big API change...